lib/steps/step_list.go

```diff
@@ -48,3 +55,34 @@ func (stepList *StepList) Prepend(step Step) {
 func (stepList *StepList) PrependList(otherList StepList) {
 	stepList.List = append(otherList.List, stepList.List...)
 }
+
+// WrapOptions represents the options given to Wrap.
+type WrapOptions struct {
+	RunInGitRoot     bool
+	StashOpenChanges bool
+}
+
+// Wrap wraps the given StepList in steps that
+// change to the Git root directory or stash away open changes.
+func (stepList *StepList) Wrap(options WrapOptions) {
```

Comment:
I don't like the `Steps` function as it seems like a getter function but will also morph the state or be recreating those wrapper steps each time. I like keeping the setter / getters separate.

We could also use an initializer function and create those steps upon construction but then we lose the ability to use our prepend / append functions as we would be operating on `[]Step`